packages/dd-trace/src/profiling/exporters/agent.js

```diff
@@ -1,11 +1,58 @@
 'use strict'
 
+const http = require('http')
 const FormData = require('form-data')
 
+function wait (ms) {
+  return new Promise((resolve) => setTimeout(resolve, ms))
+}
+
+function * times (base, timeout) {
+  let attempt = 0
+  let n
+
+  // base * n is the current stride length
+  // base * (n - 1) is the sum prior steps
+  // therefore, base * n + base * (n - 1) is the final total
+  while (timeout > base * ((n = 2 ** attempt++) - 1) + base * n) {
+    yield base * n * Math.random()
+  }
+}
+
+async function backoff (base, overallTimeout, task) {
+  for (const timeout of times(base, overallTimeout)) {
+    const start = Date.now()
+    if (await task(timeout)) return
+    const remaining = timeout - (Date.now() - start)
+    if (remaining > 0) await wait(remaining)
+  }
+
+  throw new Error('Profiler agent export back-off failed')
+}
+
+function request (options, body) {
+  return new Promise((resolve, reject) => {
+    const request = http.request(options, resolve)
```

Comment:
The request function should be cached at the import, otherwise it will be instrumented by the tracer. This is probably already the case which would be a bug.